Installing with `operator.configAuditScannerEnabled=false` in helm causes trivy-operator to crash

**What steps did you take and what happened:**

Installing with `operator.configAuditScannerEnabled=false` in helm causes trivy-operator to crash.

```shell
$ helm install trivy-operator aqua/trivy-operator --namespace trivy-system --create-namespace --set="operator.configAuditScannerEnabled=false" --version 0.11.0

$ kubectl get pods
NAME                              READY   STATUS             RESTARTS     AGE
trivy-operator-6f845fb77c-zxhn2   0/1     CrashLoopBackOff   1 (4s ago)   6s

$ kubectl logs trivy-operator-6f845fb77c-zxhn2
{"level":"info","ts":"2023-02-06T04:44:30Z","logger":"main","msg":"Starting operator","buildInfo":{"Version":"0.11.0","Commit":"23b2914f793936c32d16abbeb457802d53d375b3","Date":"2023-02-05T10:42:09Z","Executable":""}}
{"level":"info","ts":"2023-02-06T04:44:30Z","logger":"operator","msg":"Resolved install mode","install mode":"AllNamespaces","operator namespace":"trivy-system","target namespaces":[],"exclude namespaces":"","target workloads":["pod","replicaset","replicationcontroller","statefulset","daemonset","cronjob","job"]}
{"level":"info","ts":"2023-02-06T04:44:30Z","logger":"operator","msg":"Watching all namespaces"}
{"level":"info","ts":"2023-02-06T04:44:31Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1973d28]

goroutine 1 [running]:
github.com/aquasecurity/trivy-operator/pkg/operator.Start({_, _}, {{_, _}, {_, _}, {_, _}, {_, _}}, ...)
        /home/runner/work/trivy-operator/trivy-operator/pkg/operator/operator.go:212 +0x1908
main.run()
        /home/runner/work/trivy-operator/trivy-operator/cmd/trivy-operator/main.go:50 +0x230
main.main()
        /home/runner/work/trivy-operator/trivy-operator/cmd/trivy-operator/main.go:34 +0x1c
```

`GetConfigAuditPlugin()` returns `nil` for `pluginContext` because `configAuditReports.scanner` does not exist in `trivy-operator` configMap.

- https://github.com/aquasecurity/trivy-operator/blob/v0.11.0/pkg/operator/operator.go#L212
- https://github.com/aquasecurity/trivy-operator/blob/v0.11.0/pkg/plugins/factory.go#LL78-L94

```shell
$ kubectl get configmap trivy-operator -o yaml
apiVersion: v1
data:
  report.recordFailedChecksOnly: "true"
  scanJob.compressLogs: "true"
  scanJob.podTemplateContainerSecurityContext: '{"allowPrivilegeEscalation":false,"capabilities":{"drop":["ALL"]},"privileged":false,"readOnlyRootFilesystem":true}'
  vulnerabilityReports.scanner: Trivy
kind: ConfigMap
metadata:
  annotations:
    meta.helm.sh/release-name: trivy-operator
    meta.helm.sh/release-namespace: trivy-system
  creationTimestamp: "2023-02-06T04:44:29Z"
  labels:
    app.kubernetes.io/instance: trivy-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: trivy-operator
    app.kubernetes.io/version: 0.11.0
    helm.sh/chart: trivy-operator-0.11.0
  name: trivy-operator
  namespace: trivy-system
  resourceVersion: "3696279"
  uid: 2941be0e-6f75-4054-935f-3f82fdaddcf7
```

**What did you expect to happen:**

If `operator.configAuditScannerEnabled=false`, either skip the call to `GetConfigAuditPlugin()` or add `configAuditReports.scanner: Trivy` to configMap by default at helm install.


**Environment:**

- Trivy-Operator version (use `trivy-operator version`): 0.11.0
- Kubernetes version (use `kubectl version`): Client 1.26.1 / Server 1.25.0
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): Client macOS Monterey 12.6.3 / Server Ubuntu 22.04

--- a/pkg/operator/operator.go
+++ b/pkg/operator/operator.go
@@ -201,24 +201,27 @@ func Start(ctx context.Context, buildInfo trivyoperator.BuildInfo, operatorConfi
 		}
 
 		if operatorConfig.ScannerReportTTL != nil {
-			plugin, pluginContext, err := plugins.NewResolver().WithBuildInfo(buildInfo).
-				WithNamespace(operatorNamespace).
-				WithServiceAccountName(operatorConfig.ServiceAccount).
-				WithConfig(trivyOperatorConfig).
-				WithClient(mgr.GetClient()).
-				WithObjectResolver(&objectResolver).
-				GetConfigAuditPlugin()
-			if err != nil {
-				return fmt.Errorf("initializing %s plugin: %w", pluginContext.GetName(), err)
+			ttlReconciler := &TTLReportReconciler{
+				Logger: ctrl.Log.WithName("reconciler").WithName("ttlreport"),
+				Config: operatorConfig,
+				Client: mgr.GetClient(),
+				Clock:  ext.NewSystemClock(),
 			}
-			if err = (&TTLReportReconciler{
-				Logger:         ctrl.Log.WithName("reconciler").WithName("ttlreport"),
-				Config:         operatorConfig,
-				PluginInMemory: plugin,
-				PluginContext:  pluginContext,
-				Client:         mgr.GetClient(),
-				Clock:          ext.NewSystemClock(),
-			}).SetupWithManager(mgr); err != nil {
+			if operatorConfig.ConfigAuditScannerEnabled {
+				plugin, pluginContextCofig, err := plugins.NewResolver().WithBuildInfo(buildInfo).
+					WithNamespace(operatorNamespace).
+					WithServiceAccountName(operatorConfig.ServiceAccount).
+					WithConfig(trivyOperatorConfig).
+					WithClient(mgr.GetClient()).
+					WithObjectResolver(&objectResolver).
+					GetConfigAuditPlugin()
+				if err != nil {
+					return fmt.Errorf("initializing %s plugin: %w", pluginContext.GetName(), err)
+				}
+				ttlReconciler.PluginContext = pluginContextCofig
+				ttlReconciler.PluginInMemory = plugin
+			}
+			if err = ttlReconciler.SetupWithManager(mgr); err != nil {
 				return fmt.Errorf("unable to setup TTLreport reconciler: %w", err)
 			}
 		}

--- a/pkg/operator/ttl_report.go
+++ b/pkg/operator/ttl_report.go
@@ -113,6 +113,9 @@ func (r *TTLReportReconciler) applicableForDeletion(report client.Object, ttlRep
 	if ttlReportAnnotationStr == "0h0m0s" { // check if it marked as historical report
 		return true
 	}
+	if !r.Config.ConfigAuditScannerEnabled {
+		return false
+	}
 	resourceKind, ok := report.GetLabels()[trivyoperator.LabelResourceKind]
 	if !ok {
 		return false

